### ISSUE TITLE
Bug: Can't export only nodes content using exportDOM

### ISSUE THREAD

**Issue by matisszemturis** [facebook/lexical#5192]
I created a cusom node that is used as "SelectionNode" for persisting active selection in editor at all times.
<img width="496" alt="image" src="https://github.com/facebook/lexical/assets/50548714/e7212a08-9b9d-42c8-a195-023a9fe64432">
This nodes **styling** should not be exported in to HTML.

I have tried multiple techniques for exportDOM
Diggin in DOM, childNodes[0] or firstChild returns undefined
`const {element} = super.exportDOM(editor)`
`element.firstChild` -> undefined

This also just removed the node completely or crashes editor
`return { element: null }`, `return {}`, `return null`

Lexical version:
0.12.2

## Steps To Reproduce
1. Open sandbox
2. Write text
3. Select part of text
4. Click "Toggle Selection"
5. Text gets highlighted
6. HTML export should look the same as it was after step 3

Link to code example:
https://codesandbox.io/s/lexical-plain-text-example-forked-jmxl95?file=/src/nodes/SelectionNode.js

## The current behavior
<img width="525" alt="image" src="https://github.com/facebook/lexical/assets/50548714/8f3e9b02-88c2-4723-8a13-c98600a73903">

## The expected behavior
<img width="443" alt="image" src="https://github.com/facebook/lexical/assets/50548714/94b99fbe-402e-442c-9ad6-afdec098693d">

**Comment by acywatson** [comment]
I really thought there was a way to do this. Thanks for reporting.

**Comment by matisszemturis** [comment]
@acywatson Is there anything that could be done to speed up this bugs resolving?

**Comment by matisszemturis** [comment]
For now, i created a this quick fix to merge sibling span elements together. This is not optimal, because it is not implemented in $appendNodesToHTML directly, but does another loop through HTML after Lexical.
`function mergeSiblingSpans(element: HTMLElement) {
  const children = element.childNodes
  const isElement = element.nodeType === Node.ELEMENT_NODE
  if (isElement) {
    let span = null

    for (let i = 0; i < children.length; i++) {
      const child = children[i] as HTMLElement
      const nextChild = children?.[i + 1] as HTMLElement
      // console.log(child)
      if (child?.nodeName === 'SPAN' && nextChild?.nodeName === 'SPAN') {
        child.innerHTML += nextChild.innerHTML
        nextChild.remove()
        i--
      } else {
        mergeSiblingSpans(child)
      }
    }
  }
}
`

**Comment by etrepum** [comment]
I think this would be tricky to do since the logic to merge text nodes is baked into the reconciler, basically like a node transform (see `$normalizeTextNode`). The functionality is not exported anywhere, and it works at the lexical node and not the DOM node level. I'm not sure it really makes sense to try and do this separately in `$generateNodesForHTML` for this use case.

I think there are three approaches that may work well enough, in the order of how well they would likely work (best first):

* Don't do this at all - keep the selection tracking out of the document (e.g. the new `selectionAlwaysOnDisplay` functionality)
* Create a new headless editor with the existing editorState, add a transform or just do an update for MarkNode to remove themselves, and then use that for serialization. This moves the responsibility out of `exportDOM` altogether since they won't be there to export and the text nodes will be merged in that reconciliation cycle.
* Implement this normalization as you've done (knowing that it's a bit error-prone when formatting is involved since these tags then can get nested, e.g. `<sup><strong>` when they are exported to DOM)

**Comment by LeSingh1** [comment]
Checked this against current `main`. The `excludeFromCopy` part itself works, and has since #2246: `@lexical/html`'s `$appendNodesToHTML` consults `$shouldExclude` → `ElementNode.excludeFromCopy('html')`, and an excluded element is dropped while its children are appended in its place. Reproducing with a `MarkNode` around part of a text run, the export contains no wrapper tag at all:

```
<p><span style="white-space: pre-wrap;">This</span><span style="white-space: pre-wrap;"> is </span><span style="white-space: pre-wrap;">heading</span></p>
```

So what remains is not the wrapper — it is that wrapping split one `TextNode` into three, and `$generateHtmlFromNodes` emits one `<span>` per `TextNode`. That is exactly what @etrepum addressed above: text-node merging lives in the reconciler (`$normalizeTextNode`), operates on Lexical nodes rather than DOM nodes, and is not exported, so re-implementing it inside the export walk would be a separate and error-prone mechanism — nested formatting becomes nested tags, which no longer merge by tag name.

Two of the three suggested workarounds are easier now than when they were written:

- If the goal is only to keep the selection visible, `selectionAlwaysOnDisplay` is exported from `@lexical/utils` and paints the selection without putting any node in the document, so the exported HTML is untouched by construction.
- If the marks must live in the document, export from a headless editor seeded with the same `EditorState`, remove the mark nodes in an update, and serialize that. The removal reconciles first, so the adjacent text nodes are merged by `$normalizeTextNode` before `exportDOM` runs — single-`<span>` output, no DOM-level merging.

Given that, I don't think there is a fix to make in `$generateHtmlFromNodes`, and this may be worth closing in favour of the `selectionAlwaysOnDisplay` route.
